CONTRIBUTING.md

```diff
@@ -33,24 +33,30 @@ cd snowpark-python
 
 #### Install the library in edit mode and install its dependencies
 
-- Create a new Python virtual environment with any Python version that we support. Currently supported version is **Python 3.8, Python 3.9, Python 3.10**. Support for Python 3.11 is available as a preview feature to all accounts. For example,
-
-  ```bash
-  conda create --name snowpark-dev python=3.8
-  ```
+- Create a new Python virtual environment with any Python version that we support.
+  - The Snowpark Python API supports **Python 3.8, Python 3.9, Python 3.10, and Python 3.11**. 
+  - The Snowpark pandas API supports **Python 3.9, Python 3.10, and Python 3.11**. Additionally, Snowpark pandas requires **Modin 0.28.1** and **pandas 2.2.1**.
 
+    ```bash
+    conda create --name snowpark-dev python=3.9
+    ```
+    
 - Activate the new Python virtual environment. For example,
 
   ```bash
   conda activate snowpark-dev
   ```
 
 - Go to the cloned repository root folder.
-- Install the Snowpark API in edit/development mode.
+  - To install the Snowpark Python API in edit/development mode, use:
 
-    ```bash
-    python -m pip install -e ".[development, pandas]"
-    ```
+      ```bash
+      python -m pip install -e ".[development, pandas]"
+      ```
+  - To install the Snowpark pandas API in edit/development mode, use:
+      ```bash
+      python -m pip install -e ".[modin-development]"
```

Comment:
in fact, this gives develop requirement for for snowpark python API and snowpark pandas API, right?  Can you make it clear here? and maybe add some description about the pandas library difference here?

This is little bit wired, I would expect set up modin development will require some extra library installation, but modin-development and development seems duplicate many of the required library

Reply:
I think it's fine to just have this section as-is, I think it would actually be less confusing for users who want to contribute to Snowpark pandas just to have this one install and not worry too much about the details.